src/zepben/evolve/services/network/tracing/networktrace/actions/equipment_tree_builder.py

```diff
@@ -35,6 +35,7 @@ class EquipmentTreeBuilder(StepActionWithContextValue):
     """
 
     _roots: dict[ConductingEquipment, EquipmentTreeNode] = {}
+    leaves: dict[str, EquipmentTreeNode] = {}
```

Comment:
Why does this need to be a dict? Once you follow the other comment and turn it into a property, I think it should just return a Set of EquipmentTreeNode. From the users perspective they'll never know what the leaves are beforehand.

Reply:
fixed